pyopensprinkler/__init__.py

```diff
@@ -1,84 +1,165 @@
 """Main OpenSprinkler module."""
 
+import hashlib
 import json
 import urllib
 
 import httplib2
 from backoff import expo, on_exception
 
-from pyopensprinkler.device import Device
 from pyopensprinkler.program import Program
 from pyopensprinkler.station import Station
 
-_HTTP = httplib2.Http()
 
+class Controller(object):
+    """OpenSprinkler Controller"""
 
-class OpenSprinkler(object):
-    """Represent the OpenSprinkler API."""
+    def __init__(self, url, password, client_opts=None):
+        """OpenSprinkler Controller initializer."""
 
-    def __init__(self, host, md5password):
-        """OpenSprinkler class initializer."""
-        self._host = host
-        self._md5password = md5password
-        self._baseUrl = f"http://{self._host}"
-        self._programs = []
-        self._stations = []
+        if client_opts is None:
```

Comment:
Just wondering what the difference is with this and having `clients_opts={}` in the constructor signature?

Reply:
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Crazy python stuff.

Reply:
Well, I learned something today